src/commands/openYAMLConfigFile.ts

```diff
@@ -47,12 +47,13 @@ export async function openYAMLConfigFile(context: IActionContext, node?: StaticW
 
 async function getSelection(configDocument: TextDocument, buildConfigToSelect: BuildConfig): Promise<Range | undefined> {
     const configRegex: RegExp = new RegExp(`${buildConfigToSelect}:`);
+    const offset: number = configDocument.getText().search(configRegex);
 
-    let offset: number = configDocument.getText().search(configRegex);
-    // Shift the offset to the beginning of the build config's value
-    offset += `${buildConfigToSelect}: `.length;
+    if (offset === -1) {
+        return undefined;
+    }
 
     const position: Position = configDocument.positionAt(offset);
-    const configValueRegex: RegExp = /['"].*['"]/;
-    return configDocument.getWordRangeAtPosition(position, configValueRegex);
+    const line: TextLine = configDocument.lineAt(position);
+    return line.range;
```

Comment:
We might continue to have edge-case bugs with this unless we use a yaml parser. Can you double check to see if `yaml` or `js-yaml` (which seems to be more widely used) gives us the position info in one of their methods? You might have to explore outside the default "parse" method.